Makefile

```diff
@@ -1,27 +1,96 @@
-FEATURES_INTERNALS=--features internals
+.DEFAULT_GOAL := help
+
+.PHONY: help
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+# -- variables --------------------------------------------------------------------------------------
+
+WARNINGS=RUSTDOCFLAGS="-D warnings"
+DEBUG_ASSERTIONS=RUSTFLAGS="-C debug-assertions"
 FEATURES_CONCURRENT_EXEC=--features concurrent,executable
 FEATURES_LOG_TREE=--features concurrent,executable,tracing-forest
 FEATURES_METAL_EXEC=--features concurrent,executable,metal
-PROFILE_OPTIMIZED=--profile optimized
-PROFILE_TEST=--profile test-release
 
-bench:
-	cargo bench $(PROFILE_OPTIMIZED)
+# -- linting --------------------------------------------------------------------------------------
+
+.PHONY: clippy
+clippy: ## Runs Clippy with configs
+	cargo +nightly clippy --workspace --all-targets --all-features -- -D warnings
+
+
+.PHONY: fix
+fix: ## Runs Fix with configs
+	cargo +nightly fix --allow-staged --allow-dirty --all-targets --all-features
+
+
+.PHONY: format
+format: ## Runs Format using nightly toolchain
+	cargo +nightly fmt --all
+
+
+.PHONY: format-check
+format-check: ## Runs Format using nightly toolchain but only in check mode
+	cargo +nightly fmt --all --check
+
+
+.PHONY: lint
+lint: format fix clippy ## Runs all linting tasks at once (Clippy, fixing, formatting)
+
+# --- testing -------------------------------------------------------------------------------------
+
+.PHONY: test
+test: ## Runs all tests
+	$(DEBUG_ASSERTIONS) cargo nextest run --cargo-profile test-release --features internals
```

Comment:
Could we move the "testing" section after "docs". I think that's the order we have in `miden-base`.

Also, we are missing `check` target (which runs `cargo check`). I'd add it here to keep things consistent with `miden-base`.